app/packages/core/src/plugins/SchemaIO/components/TimelineView.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { Timeline, useCreateTimeline, useTimeline } from "@fiftyone/playback";
+import { ViewPropsType } from "../utils/types";
+
+export default function TimelineView(props: ViewPropsType) {
+  const { schema } = props;
+  const { view = {} } = schema;
+  const { timeline_name, loop, total_frames } = view;
+
+  const providedcConfig = {
+    loop,
+    totalFrames: total_frames,
+  };
+
+  const defaultConfig = {
+    loop: false,
+  };
+  const finalConfig = {
+    ...defaultConfig,
+    ...providedcConfig,
+  };
+
+  if (!timeline_name) {
+    throw new Error("Timeline name is required");
+  }
+  if (!finalConfig.totalFrames) {
+    throw new Error("Total frames is required");
+  }
+
+  return <TimelineCreator timelineName={timeline_name} {...finalConfig} />;
+}
+
+export const TimelineCreator = ({ timelineName, totalFrames, loop }) => {
+  const { isTimelineInitialized } = useCreateTimeline({
+    name: timelineName,
+    config: {
+      totalFrames: totalFrames,
+      loop,
+    },
+  });
```

Comment:
I'd suggest memoizing either the config or the whole arg...because if you don't, it's a new ref passed to the hook in every render. (I do have a mechanism in place to check for that but still a good practice when you're passing objects to hooks to memoize them). So you could:

```
 const timelineConfig = useMemo(() => ({
      totalFrames,
      loop,
    }), [totalFrames, loop]);

  const { isTimelineInitialized } = useCreateTimeline({
    name: timelineName,
    config: timelineConfig,
  });
```

this way `timelineConfig` is referentially stable and any downstream hooks that use it don't suffer or don't have to be super clever

Reply:
React is so amazing!